### PR TITLE
preprocess: add delay to improve changes of data being available

### PR DESCRIPTION
**Pull request by juliangruber**
As suggested in https://filecoinproject.slack.com/archives/C031K3BG527/p1714658724173009?thread_ts=1714636341.173339&cid=C031K3BG527. Data stored in W3S isn't always immediately available, and especially sometimes trying to fetch it too soon will result in bad responses in the caches.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,6 +4,10 @@ import { preprocess } from './lib/preprocess.js'
 import { evaluate } from './lib/evaluate.js'
 import { onContractEvent } from 'on-contract-event'
 import { RoundData } from './lib/round.js'
+import timers from 'node:timers/promises'
+
+// Tweak this value to improve the chances of the data being available
+const PREPROCESS_DELAY = 60_1000
```
Comment (bajtos) [inline index.js:10]: This value is 601 seconds, did you mean `60_000` instead?

60 seconds seems a bit long delay to me, how about starting with a smaller value like `10_000` or `30_000`?

**Review by bajtos — CHANGES_REQUESTED** [review]
+1 to give this a try.

There is a big catch though: by introducing this delay, when we receive the last batch of measurements for a round, we are almost certainly going to evaluate this round before we fetch the measurements for this round, effectively discarding that last batch.

The current implementation does not handle this situation well either, see https://github.com/filecoin-station/spark-evaluate/issues/64#issuecomment-1830299843

WDYT, is this the right time to bite the bullet and fix the underlying race condition common to both the old and the proposed version? Or should we defer that fix for later, to ship your fix faster?

File: index.js (hunk 1 above)
Comment (juliangruber) [reply index.js:10]: Actually 60s is the lower boundary, it was suggested to go with 1-5 minutes

Comment (juliangruber) [reply index.js]: Fixed the number 😅

**Comment by juliangruber** [comment]
> There is a big catch though: by introducing this delay, when we receive the last batch of measurements for a round, we are almost certainly going to evaluate this round before we fetch the measurements for this round, effectively discarding that last batch.

This is an unfortunate truth about this logic: It's better to pretty sure miss the last measurements batch per round, than to risk missing every batch.

> WDYT, is this the right time to bite the bullet and fix the underlying race condition common to both the old and the proposed version? Or should we defer that fix for later, to ship your fix faster?

Good question. I think it's not the right time, we have more important issues (like increasing operator rewards). If nodes stay online, it doesn't matter so much if one batch is missed, as the distribution of rewards should still be roughly the same. We should 100% get this tight eventually though.

**Review by bajtos — APPROVED** [review]
👍🏻

**Pull request merged**